design/rest-api-guideline.md

```diff
@@ -0,0 +1,476 @@
+# REST API Guideline
+
+* **Status**: Draft
+* **Author**: keycloak-team
+* **JIRA**: [KEYCLOAK-9344](https://issues.jboss.org/browse/KEYCLOAK-9344)
+
+## Abstract
+
+The Keycloak REST API Guideline provides a set of design principles and practices that should be considered by developers when designing, implementing and exposing a RESTful API. To provide the best experience for developers consuming Keycloak REST APIs, it's important API developers to follow to the principles and practices defined herein, making these APIs intuitive and easy to use.
+
+The goals of the guideline are:
+
+* Define the [overall principles](#principles) that should be taken into account when designing REST APIs for Keycloak
+
+* Define common practices and patterns for all Keycloak REST APIs such as:
+    * [Filtering](#filtering)
+    * [Pagination](#pagination)
+    * [Error Handling](#error-handling)
+    * [Documentation](#documentation)
```

Comment:
@pedroigor I would like to have this section removed and follow up on this in a different PR. Please, see: https://issues.jboss.org/browse/KEYCLOAK-10787

Reply:
I would prefer to not remove it because:

* Document is marked as `Status: Draft`. I don't see an issue if we still have TODOs in a draft.
* Having them in the guideline helps to understand much better the document structure and what we want to cover in the document